File: TAbaretata/JavaScript-Fundamentals-2015/05.Arrays/05.SelectionSorting/SelectionSort.js
/* 
Problem 5: Sorting an array means to arrange its elements in increasing order. Write a script to sort an array.
Use the "selection sort" algorithm: Find the smallest element, move it at the first position, find the smallest from the rest,
move it at the second position, etc.Hint: Use a second array
*/

var arr = [1, 3, 66, 40, 21, 25, 4, 60, 33, 28, 44];

for (var i = 0; i < arr.length - 1; i++) {
    var min = i;
    for (var j = i + 1; j < arr.length; j++) {
        if (arr[j] < arr[min]) {
            min = j;
        }
    }
    var temp = arr[i];
    arr[i] = arr[min];
    arr[min] = temp;
}
for (var i = 0; i < arr.length; i++) {
    console.log(arr[i] + ' ');
}

File: TAbaretata/CSharpPartTwo-2015/Arrays/7.SelectionSort/SelectionSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

//Sorting an array means to arrange its elements in increasing order. Write a program to sort an array.
//Use the Selection sort algorithm: Find the smallest element, move it at the first position, 
//find the smallest from the rest, move it at the second position, etc.

class SelectionSort
{
    static void Main()
    {
        Console.WriteLine("Selection sorting an array.");
        Console.Write("Enter size of the array: ");
        int n = int.Parse(Console.ReadLine());
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
        {
            Console.Write("arr[{0}] = ", i);
            arr[i] = int.Parse(Console.ReadLine());
        }
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = i + 1; j < n; j++)
            {
                if (arr[i] > arr[j])
                {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
        Console.WriteLine();
        Console.Write("Selection sorted array is: ");
        for (int i = 0; i < n; i++)
        {
            Console.Write(arr[i] + " ");
        }
        Console.WriteLine();
    }
}
